src/test/ui/self/arbitrary_self_types_pointers_and_wrappers.rs

```diff
@@ -1,3 +1,4 @@
+// run-pass
```

Comment:
Please don't move run-pass tests into ui directory, or move and make them compile-pass if they don't really require running.

Reply:
@petrochenkov We're trying to retire the `run-pass` directory altogether; It's pretty bad organization to have some tests for some feature in `test/run-pass/foo` and some in `test/ui/foo`. But yeah, make it `compile-pass` if it doesn't need to be actually run.

Reply:
@Centril 
>We're trying to retire the run-pass directory altogether

This was tried, but reverted, see https://github.com/rust-lang/rust/pull/54530.

Reply:
Seems like an unfortunate move to me. Something should be done about the current chaos in the `src/test` directory. One shouldn't have to look in different places for tests pertaining to one specific interest. That is especially annoying when writing stabilization reports and whatnot.

For now, let's move ahead with moving these specific tests.